src/sentry/static/sentry/app/utils/dates.jsx

```diff
@@ -0,0 +1,109 @@
+import moment from 'moment';
+
+import {RETENTION_DAYS} from 'app/constants';
+
+function getParser(local = false) {
+  return local ? moment : moment.utc;
+}
+
+/**
+ * Returns date "RETENTION_DAYS" ago
+ */
+export function getEarliestRetentionDate() {
+  return moment()
+    .subtract(RETENTION_DAYS, 'days')
+    .toDate();
+}
+
+/**
+ * Return a date object in local time, when given a UTC timestamp
+ */
+export function getLocalDateObject(str) {
+  return moment.utc(str).toDate();
+}
+
+/**
+ * Given a date object, format in datetime in UTC
+ * given: Tue Oct 09 2018 00:00:00 GMT-0700 (Pacific Daylight Time)
+ * returns: "2018-10-09T07:00:00.000"
+ */
+export function getUtcDateString(dateObj) {
+  return moment.utc(dateObj).format(moment.HTML5_FMT.DATETIME_LOCAL_MS);
```

Comment:
Can we drop the milliseconds? I think we don't support querying at that level of granularity.

Reply:
eecab2b09b5cd